### PR TITLE
Add standard error response

### PR DESCRIPTION
**Pull request by ildarusmanov**
Introduce `rails_api_format` gem:
* Devise: 401
* Rails public exceptions: 404, 500, etc
* Rails Responder: ActiveRecord errors

```js
{
  'error': {
    'status': 404,
    'error': 'Not Found'
  }
}
```

Closes: https://github.com/fs/rails-base-api/issues/77, https://github.com/fs/rails-base-api/issues/88

@vast @VladimirMikhailov

### CODE REVIEW

File: lib/rails_api_format/lib/rails_api_format/error.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+module RailsApiFormat
+  class Error
+    include ActiveModel::Model
+    include ActiveModel::Serialization
+
+    ATTRIBBUTES = %i(status error validations)
+    attr_accessor(*ATTRIBBUTES)
```
Comment (VladimirMikhailov) [inline lib/rails_api_format/lib/rails_api_format/error.rb]: it seems like we use `ATTRIBUTES` only for defining attr accessors or have I missed something?

File: doc/api/sessions/sign_in_with_valid_password.json [hunk 2]
```diff
@@ -16,23 +16,35 @@
...
       "request_content_type": "application/x-www-form-urlencoded",
       "response_status": 201,
       "response_status_text": "Created",
-      "response_body": "{\"id\":1,\"authentication_token\":\"w3hSZvxJhansZ6bCss3y\",\"email\":\"user1@example.com\"}",
-      "response_headers": {"X-Frame-Options":"SAMEORIGIN","X-XSS-Protection":"1; mode=block","X-Content-Type-Options":"nosniff","X-UA-Compatible":"chrome=1","Content-Type":"application/json; charset=utf-8","ETag":"\"2b5b80af5b20c18e241900c6fae62462\"","Cache-Control":"max-age=0, private, must-revalidate","X-Request-Id":"8d421aac-9c05-43a7-a5f7-6daed1d5b02b","X-Runtime":"0.072063","Content-Length":"82"},
+      "response_body": "{\"user\":{\"id\":1,\"authentication_token\":\"wvEUWJZTjsevNTyPMGb1\",\"email\":\"user1@example.com\"}}",
+      "response_headers": {
+        "X-Frame-Options": "SAMEORIGIN",
+        "X-XSS-Protection": "1; mode=block",
+        "X-Content-Type-Options": "nosniff",
```
Comment (VladimirMikhailov) [inline doc/api/sessions/sign_in_with_valid_password.json]: The `X-Frame-Options` and `X-XSS-Protection` are mostly used by browser policies, should we get rid of them in the api version?

**Comment by VladimirMikhailov** [comment]
And the rest looks great, thank you

File: lib/rails_api_format/lib/rails_api_format/error.rb (hunk 1 above)
Comment (timurvafin) [inline lib/rails_api_format/lib/rails_api_format/error.rb]: Fixed in https://github.com/fs/rails-base-api/commit/570f2cc77243c03a7548d7fb1046e8fbc02ea79c

File: doc/api/sessions/sign_in_with_valid_password.json (hunk 2 above)
Comment (timurvafin) [inline doc/api/sessions/sign_in_with_valid_password.json]: Right now I've just removed them from documentation https://github.com/fs/rails-base-api/pull/177
But for sure we could remove them from response too, I've created separate issue for that https://github.com/fs/rails-base-api/issues/178

**Pull request merged**